### PR TITLE
MGMT-8824: Copy additional NTP sources to chrony manifests

### PR DESCRIPTION
**Pull request by mresvanis**
This PR addresses issue [MGMT-8824](https://issues.redhat.com/browse/MGMT-8824). 

#### High-level bug description

When a user defined [additional NTP sources](https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L6976) (or via the [InfraEnv](https://github.com/openshift/assisted-service/blob/master/api/v1beta1/infraenv_types.go#L57) CR), only the sources that were in state `synchronized|synced` were copied into the respective [chrony](https://chrony.tuxfamily.org/) manifest, if any. Which resulted in the additional NTP sources missing from the chrony sources altogether. 

#### Fix 

With this change, all user defined additional NTP sources, irrespective of their state, are copied into the chrony manifest and are reflected in the chrony sources.

> The possible source states can also be found [here](https://chrony.tuxfamily.org/doc/3.3/chronyc.html), under the section `Time Sources`.

> To check the chrony sources of a host, you can use the [chronyc](https://chrony.tuxfamily.org/doc/4.2/chronyc.html) `sources` command: `chronyc sources -c -v`

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [x] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by carbonin** [comment]
Is there any side effect to this?
What does it mean to ignore this state?

**Comment by carbonin** [comment]
Otherwise it looks good to me.

File: internal/network/manifests_generator.go [hunk 1]
```diff
@@ -190,12 +190,8 @@ func createChronyManifestContent(c *common.Cluster, role models.HostRole, log lo
 		}
 
 		for _, source := range ntpSources {
```
[the remark this answers is not in the document]
Comment (mresvanis) [reply internal/network/manifests_generator.go:192]: Good point. As far as I could see, the flow that results in the `ntpSources` slice is the following:

* the assisted-service [fetches the host's NTP sources](https://github.com/openshift/assisted-service/blob/master/internal/host/hostcommands/ntp_synchronizer_cmd.go#L29) from the [assisted-install-agent ntp-synchronizer cmd](https://github.com/openshift/assisted-installer-agent/blob/master/src/ntp_synchronizer/ntp_synchronizer.go#L154)

* the payload of the previous request includes the additional NTP sources (if any), which triggers their [addition to chrony](https://github.com/openshift/assisted-installer-agent/blob/master/src/ntp_synchronizer/ntp_synchronizer.go#L164) by the assisted-install-agent ntp-synchronizer

* the host's NTP sources now include the set of the chrony sources already on the host + the additional NTP sources defined by the user

If I understand your question correctly, the `host.NtpSources` should include the `additional_ntp_source` and this is the set from which we are generating the chrony manifest.

**Comment by mresvanis** [comment]
@carbonin 

> Is there any side effect to this? What does it mean to ignore this state?

### TL;DR
There should be no side effect by ignoring the additonal NTP sources' state, which are going to be included in the chrony manifest.

### Explanation
 
The state of each NTP source is reported via chrony by the [ntp-synchronizer](https://github.com/openshift/assisted-installer-agent/blob/master/src/ntp_synchronizer/ntp_synchronizer.go#L94). The state `synchronized|synced` defines an NTP source that is currently the best source selected for synchronization. That does not mean however that the other sources are invalid, but rather that there is a better suited server that is currently selected. 

By letting the users define any additional NTP sources, it would mean that if they have defined a better suited NTP source, it will probably be selected. And if they have defined a less suited server that it probably won't be selected. In any case though, all sources can be safely on the chrony configuration. 

> But I am by no means really familiar with NTP and chrony specifically, so maybe I'm missing something :)

**Comment by filanov** [comment]
Overall LGTM, I will let Nir to take another look

**Comment by filanov** [comment]
@mresvanis can you please add more description to the commit message? 
the format usually is the first line is as you did it and then you add additional description and details below that will contain information similar to the description that you added to the PR

**Comment by mresvanis** [comment]
@filanov

>  can you please add more description to the commit message? the format usually is the first line is as you did it and then you add additional description and details below that will contain information similar to the description that you added to the PR

Very good point, I have updated the commit message to describe the change properly.

**Comment by filanov** [comment]
> @filanov
> 
> > can you please add more description to the commit message? the format usually is the first line is as you did it and then you add additional description and details below that will contain information similar to the description that you added to the PR
> 
> Very good point, I have updated the commit message to describe the change properly.

This part is not a must, but you don't need to change that. you can take a look at examples from other commits.
```
This change addresses issue MGMT-8824.

Description
-----------
```
Usually the commit message should explain the why and how to make it clear for people that are looking the history
So this is great thanks.

**Review by filanov — APPROVED** [review]

**Comment by mresvanis** [comment]
/retitle [release-ocm-2.4] Bug 2066076: Copy additional NTP sources to chrony manifests
